StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/UpdateCardViewController.swift

```diff
@@ -0,0 +1,184 @@
+        section.delegate = self
+        return section
+    }()
+
+    // MARK: Overrides
+    init(paymentOptionCell: SavedPaymentMethodCollectionView.PaymentOptionCell, paymentMethod: STPPaymentMethod, appearance: PaymentSheet.Appearance) {
+        self.paymentOptionCell = paymentOptionCell
+        self.paymentMethod = paymentMethod
+        self.appearance = appearance
+
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // disable swipe to dismiss
+        isModalInPresentation = true
+
+        let stackView = UIStackView(arrangedSubviews: [formStackView])
+        stackView.spacing = PaymentSheetUI.defaultPadding
+        stackView.axis = .vertical
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.backgroundColor = appearance.colors.background
+        view.backgroundColor = appearance.colors.background
+        view.addSubview(stackView)
+
+        NSLayoutConstraint.activate([
+            stackView.topAnchor.constraint(equalTo: view.topAnchor),
+            stackView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            stackView.centerYAnchor.constraint(equalTo: view.centerYAnchor),
+            stackView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
+            stackView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
+        ])
+    }
+
+    // MARK: Private helpers
+    private func dismiss() {
+        guard let bottomVc = parent as? BottomSheetViewController else { return }
+        _ = bottomVc.popContentViewController()
+    }
+
+}
+
+// MARK: BottomSheetContentViewController
+extension UpdateCardViewController: BottomSheetContentViewController {
+
+    var allowsDragToDismiss: Bool {
+        return false
+    }
+
+    func didTapOrSwipeToDismiss() {
+        // no-op
+    }
+
+    var requiresFullScreen: Bool {
+        return false
+    }
+}
+
+// MARK: SheetNavigationBarDelegate
+extension UpdateCardViewController: SheetNavigationBarDelegate {
+
+    func sheetNavigationBarDidClose(_ sheetNavigationBar: SheetNavigationBar) {
+        dismiss()
+    }
+
+    func sheetNavigationBarDidBack(_ sheetNavigationBar: SheetNavigationBar) {
+        dismiss()
+    }
+
+}
+
+// MARK: ElementDelegate
+extension UpdateCardViewController: ElementDelegate {
+    func continueToNextField(element: Element) {
+        // no-op
+    }
+
+    func didUpdate(element: Element) {
+        let selectedBrand = cardBrandDropDown.selectedItem.rawData
+        let currentCardBrand = STPCard.brand(from: paymentMethod.card?.networks?.preferred ?? "").rawValue
+        updateButton.update(state: selectedBrand != "\(currentCardBrand)" && selectedBrand != "-1" ? .enabled : .disabled)
```

Comment:
Is this converting the rawValue of the preferred card brand to a String? And why `-1` here?

Reply:
Good feedback! I switch this from string comparison to STPCardBrand comparison.